mypy/server/deps.py

```diff
@@ -240,8 +239,9 @@ def visit_class_def(self, o: ClassDef) -> None:
             self.is_class = old_is_class
 
     def visit_newtype_expr(self, o: NewTypeExpr) -> None:
-        with self.scope.class_scope(o.info) if o.info else nullcontext():
-            self.process_type_info(o.info)
+        if o.info:
```

Comment:
What are the code changes needed for? This one seems like a real logic change. (Though a correct one, I suspect!)

Reply:
it's a revert to https://github.com/python/mypy/commit/725a24a82fdf79521db5d0c43549e8731f793c55

Reply:
Right, but why?

Reply:
well, it didn't pass the type check since the type of o.info is optional